python/src/etos_api/routers/v1alpha/utilities.py

```diff
@@ -0,0 +1,179 @@
+    :type timeout: int
+    :return: ArtifactCreated edges from GraphQL.
+    :rtype: list
+    """
+    timeout = time.time() + timeout
+    query_handler = GraphqlQueryHandler(etos_library)
+    if artifact_id is not None:
+        LOGGER.info("Verify that artifact ID %r exists.", artifact_id)
+        query = VERIFY_ARTIFACT_ID_EXISTS
+    elif artifact_identity is not None:
+        LOGGER.info("Getting artifact from packageURL %r", artifact_identity)
+        query = ARTIFACT_IDENTITY_QUERY
+        if artifact_identity.startswith("pkg:"):
+            # This makes the '$regex' query to the event repository more efficient.
+            artifact_identity = f"^{artifact_identity}"
+    else:
+        raise ValueError("'artifact_id' and 'artifact_identity' are both None!")
+    artifact_identifier = artifact_identity or str(artifact_id)
+
+    LOGGER.debug("Wait for artifact created event.")
+    while time.time() < timeout:
+        try:
+            artifacts = await query_handler.execute(query % artifact_identifier)
+            assert artifacts is not None
+            assert artifacts["artifactCreated"]["edges"]
+            return artifacts["artifactCreated"]["edges"]
+        except (AssertionError, KeyError):
+            LOGGER.warning("Artifact created not ready yet")
+        await asyncio.sleep(2)
+    LOGGER.error("Artifact %r not found.", artifact_identifier)
+    return None
+
+
+async def download_suite(test_suite_url: str) -> list[dict[str, Any]]:
+    """Attempt to download suite.
+
+    :param test_suite_url: URL to test suite to download.
+    :return: Downloaded test suite as JSON.
+    """
+    try:
+        suite = requests.get(test_suite_url, timeout=60)
+        suite.raise_for_status()
+    except Exception as exception:  # pylint:disable=broad-except
+        raise AssertionError(f"Unable to download suite from {test_suite_url}") from exception
+    return suite.json()
+
+
+async def validate_suite(test_suite: list[dict[str, Any]]) -> None:
+    """Validate the ETOS test suite through the SuiteValidator.
+
+    :param test_suite_url: The URL to the test suite to validate.
+    """
+    span = trace.get_current_span()
+
+    try:
+        await SuiteValidator().validate(test_suite)
+    except AssertionError as exception:
+        LOGGER.error("Test suite validation failed!")
+        LOGGER.error(exception)
+        span.add_event("Test suite validation failed")
+        raise HTTPException(
+            status_code=400, detail=f"Test suite validation failed. {exception}"
+        ) from exception
+
+
+def convert_to_rfc1123(value: str) -> str:
+    """Convert string to RFC-1123 accepted string.
+
+    https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names
+
+    Some resource types require their names to follow the DNS label standard as defined in RFC 1123.
+    This means the name must:
+
+        contain at most 63 characters
```

Comment:
Make sure requirements are checked or enforced by the function.

Reply:
Reading just below
`This method does not care about the length of the string since ETOS uses generateName for creating Kubernetes resources and that function will truncate the string down to 63-5 and then add 5 random characters.`

Also, I did not change the code I just moved it.